google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -700,6 +867,36 @@ def _get_initial_request(
         # Return the initial request.
         return request
 
+    def _send_lease_modacks(self, ack_ids: Sequence[str], ack_deadline: int):
+        exactly_once_enabled = False
+        with self._exactly_once_enabled_lock:
+            exactly_once_enabled = self._exactly_once_enabled
+        if exactly_once_enabled:
+            items = []
+            for ack_id in ack_ids:
+                future = futures.Future()
+                request = requests.ModAckRequest(ack_id, ack_deadline, future)
+                items.append(request)
+
+            assert self._dispatcher is not None
+            self._dispatcher.modify_ack_deadline(items)
+
+            for req in items:
+                try:
+                    req.future.result()
+                except AcknowledgeError:
+                    _LOGGER.warning(
+                        "AcknowledgeError when modacking a message immediately after receiving it.",
```

Comment:
Is this comment correct? Looks like this code will be executed more often than just after receiving the message (and for continuous lease extension)

Reply:
My mistake, fixed.